lib/spack/spack/compilers/__init__.py

```diff
@@ -125,6 +125,105 @@ def get_compiler_config(scope=None, init_config=True):
     return config
 
 
+def get_compiler_config_from_packages(scope=None):
+    """Return the compiler configuration from packages.yaml"""
+    packages = []
+    config = spack.config.get("packages", scope=scope)
```

Comment:
(minor comment) I think it is better if we start injecting globals into functions, etc. in particular for functions that are not widely called. For instance:
```suggestion
def get_compiler_config_from_packages(*, configuration, scope=None):
    """Return the compiler configuration from packages.yaml"""
    packages = []
    config = configuration.get("packages", scope=scope)
```
That makes it easier to find where e.g. `spack.config.CONFIG` is used, and to unit test the function.